pkg/KubeArmorController/main.go

```diff
@@ -139,29 +143,29 @@ func main() {
 }
 
 // detect the enforcer on the node
-func detectEnforcer(logger logr.Logger) string {
+func detectEnforcer(logger logr.Logger, client client.Client, nodeName string) string {
 	// assumption: all nodes have the same OSes
 
-	lsm := []byte{}
-	lsmPath := "/sys/kernel/security/lsm"
-
-	if _, err := os.Stat(filepath.Clean(lsmPath)); err == nil {
-		lsm, err = os.ReadFile(lsmPath)
-		if err != nil {
-			logger.Info("Failed to read /sys/kernel/security/lsm " + err.Error())
-			return ""
-		}
+	node := &v1.Node{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, node)
+	if err != nil {
+		logger.Error(err, "Failed to get node information")
```

Comment:
Let's handle the cases where we miss getting information from some specific node in a big cluster. 
```suggestion
		logger.Error(err, "Failed to get enforcer information from node", "nodeName", node.Name)
```